backend/review/views.py

```diff
@@ -125,39 +124,95 @@ def extra_metrics_section_filters_pcr(current_semester=None):
+                    {
+                        "name": "semester",
+                        "in": "query",
+                        "description": "Optionally specify the semester of the desired course (defaults to most recent course with the specified course code).",  # noqa E501
+                        "schema": {"type": "string"},
+                        "required": False,
+                    },
+                ]
+            },
+        },
         override_response_schema=course_reviews_response_schema,
     )
 )
 @permission_classes([IsAuthenticated])
-def course_reviews(request, course_code):
+def course_reviews(request, course_code, semester=None):
     """
     Get all reviews for the topic of a given course and other relevant information.
     Different aggregation views are provided, such as reviews spanning all semesters,
     only the most recent semester, and instructor-specific views.
     """
     try:
+        semester = request.GET.get("semester")
         course = (
-            Course.objects.filter(course_filters_pcr_allow_xlist, full_code=course_code)
+            Course.objects.filter(
+                course_filters_pcr,
+                **(
+                    {"topic__courses__full_code": course_code, "topic__courses__semester": semester}
+                    if semester
+                    else {"full_code": course_code}
+                ),
+            )
             .order_by("-semester")[:1]
-            .select_related(
-                "topic",
-                "topic__most_recent",
-                "topic__branched_from",
-                "topic__branched_from__most_recent",
+            .annotate(
+                aliases=Subquery(
+                    Course.objects.filter(primary_listing=OuterRef("primary_listing"))
+                    .exclude(id=OuterRef("id"))
+                    .values_list("full_code", flat=True)
+                ),
+                branched_from_full_code=F("topic__branched_from__most_recent__full_code"),
+                branched_from_semester=F("topic__branched_from__most_recent__semester"),
+                historical_codes=Subquery(
+                    Course.objects.filter(topic=OuterRef("topic"))
+                    .exclude(full_code=OuterRef("full_code"))
+                    .values("full_code")
+                    .annotate(semester=Max("semester"), branched_from=Value(False))
+                    .values("full_code", "semester", "branched_from")
+                ),
             )
-            .prefetch_related("topic__courses")
             .get()
         )
     except Course.DoesNotExist:
         raise Http404()
 
-    topic = course.topic
-    course = topic.most_recent
+    superseded = (
+        Course.objects.filter(
+            course_filters_pcr,
+            full_code=course_code,
+        )
+        .aggregate(max_semester=Max("semester"))
+        .get("max_semester")
+        > course.semester
+        if semester
+        else False
```

Comment:
Doesn't this code imply you are superseded if there is another course within the same lineage? But that's not how I understood the definition of superseded from earlier...

Would you mind elaborating on why this depends on `semester` and why we compare to the `max_semester`?

Reply:
`course` is always the most recent course in the topic. So superceded just determines if that full code has been offered in a strictly later semester, indicating the course has been superceded by a different topic

Reply:
if `semester` isn't specified, we always take the most recent course matching the full code no matter what so that check will always evaluate to False

Reply:
Got it makes sense.